source/MaterialXGenShader/ShaderGenerator.cpp

```diff
@@ -213,7 +208,7 @@ bool ShaderGenerator::implementationRegistered(const string& name) const
     return _implFactory.classRegistered(name);
 }
 
-ShaderNodeImplPtr ShaderGenerator::getImplementation(InterfaceElementPtr element)
+ShaderNodeImplPtr ShaderGenerator::getImplementation(InterfaceElementPtr element, const GenOptions& options)
```

Comment:
The GenOptions needs to be passed deeper into the shadergen code now. But this will be needed anyway later on when we want to support different optimization levels etc.